Netable/Netable/Netable.swift

```diff
@@ -115,36 +130,45 @@ open class Netable {
             }
         }
 
-        var urlRequest: URLRequest!
+        let task = Task {
+            do {
+                let result = try await self.request(request)
+                completion(.success(result))
+            } catch {
+                completion(.failure(error.netableError))
+            }
+        }
 
-        do {
-            let finalURL = try fullyQualifiedURLFrom(path: request.path)
-            urlRequest = URLRequest(url: finalURL)
-            urlRequest.httpMethod = request.method.rawValue
+        return task
+    }
 
-            guard finalURL.scheme?.lowercased() == "https" || finalURL.scheme?.lowercased() == "http" else {
-                self.log(.message("Only HTTP and HTTPS request are supported currently."))
-                throw NetableError.malformedURL
-            }
+    /**
+     * Create and send a new request, returning a tuple containing a reference to the task and a PassthroughSubject to monitor for results.
+     *
+     * - parameter request: The request to send, this has to extend `Request`.
+     *
+     * - returns: A tuple that contains a reference to the `Task` and a PassthroughSubject to monitor for results.
+     */
+    public func request<T: Request>(_ request: T) -> (task: Task<(), Never>, subject: PassthroughSubject<Result<T.FinalResource, NetableError>, Never>) {
+        let resultSubject = PassthroughSubject<Result<T.FinalResource, NetableError>, Never>()
 
-            if T.Parameters.self != Empty.self {
-                try urlRequest.encodeParameters(for: request, defaultEncodingStrategy: config.jsonEncodingStrategy)
+        let task = Task {
+            do {
+                let finalResource = try await self.request(request)
+                await MainActor.run {
```

Comment:
Might be a little cleaner, more Combine-idiomatic and (maybe) higher performance to define the subject as

```
let resultSubject = PassthroughSubject<Result<T.FinalResource, NetableError>, Never>().receive(on: RunLoop.main)
```

and ditch the MainActor.run

Reply:
Had to go with slightly different syntax but I think the outcome is the same ⭐